packages/client3/components/nav/wallet-button.tsx

```diff
@@ -1,43 +1,119 @@
 
-import { Identicon } from "../identicon";
-import { WalletStatus } from "./wallet-status";
+import { Button } from "../design-system";
+import { User, ButtonStateText, IWalletButtonStyles } from "./types";
 
-export function WalletButton() {
-  const { account, error, connector } = useWallet();
+export function WalletButton({
+  getAccount: getAccount,
+}: {
+  getAccount: (account: User) => void;
+}) {
+  const { account, isActive, error } = useWallet();
+  const [isLoading, setIsLoading] = useState(false);
 
-  return error ? (
-    <Button
-      className="text-clay-500 h-10"
-      iconRight="Exclamation"
-      onClick={
-        error.name === "ChainIdNotAllowedError"
-          ? () => connector.activate(DESIRED_CHAIN_ID)
-          : openWalletModal
+  const detectBrowser = useCallback(async () => {
+    const provider = await detectEthereumProvider();
+    if (provider) {
+      return true;
+    }
+    return false;
+  }, []);
+
+  const handleWalletConnect = async () => {
+    setIsLoading(true);
+    if (await detectBrowser) {
+      if (!isActive) {
+        await metaMask.activate(DESIRED_CHAIN_ID);
       }
-    >
-      {error.name === "ChainIdNotAllowedError"
-        ? "Wrong network"
-        : "Wallet error"}
-    </Button>
-  ) : account ? (
-    <Popover
-      placement="bottom-end"
-      content={({ close }) => <WalletStatus onWalletDisconnect={close} />}
-    >
-      <Button className="inline-flex h-10 items-center gap-3 !px-2 md:!px-4">
-        <span className="hidden md:block">
-          {account.substring(0, 6)}...{account.substring(account.length - 4)}
-        </span>
-        <Identicon account={account} scale={3} />
-      </Button>
-    </Popover>
+    }
+    setIsLoading(false);
+  };
+
+  function handleAddressFormat(account: string) {
+    return `${account.substring(0, 4)}...${account.substring(
+      account.length - 4
+    )}`;
+  }
+
+  const getButtonStyles = useCallback((): IWalletButtonStyles => {
+    if (isLoading) {
+      return {
+        icon: "",
+        type: ButtonType.PRIMARY,
+        state: ButtonStateText.CONNECT,
+      };
+    } else if (error) {
+      if (error.message.includes("MetaMask not installed")) {
+        return {
+          icon: "Exclamation",
+          type: ButtonType.PRIMARY,
+          state: ButtonStateText.INSTALL_METAMASK,
+          tooltip: "MetaMask not installed",
+        };
+      }
+      return {
+        icon: "Exclamation",
+        type: ButtonType.PRIMARY,
+        state: ButtonStateText.ERROR,
+        tooltip: error.message,
+      };
+    } else if (isActive) {
+      return {
+        icon: "Wallet",
+        type: ButtonType.SECONDARY,
+        state: handleAddressFormat(account as string),
+      };
+    } else {
+      return {
+        icon: "",
+        type: ButtonType.PRIMARY,
+        state: ButtonStateText.CONNECT,
+      };
+    }
+  }, [account, error, isActive, isLoading]);
```

Comment:
Sorry, I wasn't clear enough. You could avoid using useCallback in this function call as this function isn't being passed as a dependency or passed to a component.
Also,  a minor thing would be to remove `else` as there is already return in the if blocks.

This style 
```
{
        icon: "",
        type: ButtonType.PRIMARY,
        state: ButtonStateText.CONNECT,
      } 
```
seems to be returned in two conditionals, final return would be enough i think

Reply:
My bad. isLoading should've had a connecting state ("Processing....")
```
    if (isLoading) {
      return {
        icon: "",
        type: ButtonType.PRIMARY,
        state: ButtonStateText.CONNECTING,
      };
    }
```